documentation/projects/proposals/rekognition_data/20240530-implementation_plan_augment_catalog_with_rekognition_tags.md

```diff
@@ -0,0 +1,669 @@
+
+Machine-generated tags that are the product of AI image labeling models have
+been shown repeatedly and consistently to perpetuate certain cultural,
+structural, and institutional biases[^1][^2][^3]. This includes analysis done on
+[AWS Rekognition](https://docs.aws.amazon.com/rekognition/),
+specifically[^4][^5][^6].
+
+Certain demographic axes seem the most likely to result in an incorrect or
+insensitive label (e.g. gender assumption of an individual in a photo). For the
+reasons described in the above cited works, we should **exclude** labels that
+have a demographic context in the following categories:
+
+- Age
+- Gender
+- Sexual orientation
+- Nationality
+- Race
+
+There are other categories which might be useful for search relevancy and are
+less likely to be applied in an insensitive manner. These labels **should not**
+be excluded. Some examples include:
+
+- Occupation
+- Marital status
+- Health and disability status
+- Political affiliation or preference
+- Religious affiliation or preference
+
+### Accuracy selection
+
+[^removal]: Note that this step will be going away as part of #430
+
+We already filter out existing tags from the catalog when copying data into the
+API database during the data refresh's
+[cleanup step](https://github.com/WordPress/openverse/blob/3747f9aa40ed03899becb98ecae2abf926c8875f/ingestion_server/ingestion_server/cleanup.py#L119-L150)[^removal].
+The minimum accuracy value used for this step is
+[0.9 (or 90%)](https://github.com/WordPress/openverse/blob/3747f9aa40ed03899becb98ecae2abf926c8875f/ingestion_server/ingestion_server/cleanup.py#L57-L56)
+. AWS's own advice on what value to use is that
+[it depends entirely on the use case of the application](https://aws.amazon.com/rekognition/faqs/#Label_Detection).
+
+I took a small sample of the labels we have available (~100MB out of the 196GB
+dataset, about 45k images with labels) and performed some exploratory analysis
+on the data. I found the following pieces of information:
+
+- **Total images**: 45,059
+- **Total labels**: 555,718
+- **Average confidence across all labels**: 79.927835
+- **Median confidence across all labels**: 81.379463
+- **Average confidence per image**: 81.073921
+- **Median confidence per image**: 82.564148
+- **Number of labels with confidence higher than 90**: 210,341
+- **Percentage of labels with confidence higher than 90**: 37.85031%
+- **Average number of labels per image higher than 90**: 4.6629
+
+_For a full explanation on this exploration, see:
+[Analysis explanation](#analysis-explanation)_
+
+Based on the number of labels we would still be receiving with a confidence
+higher than 90, and that 0.9 is already our existing minimum standard, **we
+should retain 0.9 or 90% as our minimum label accuracy value** for inclusion in
+the catalog.
```

Comment:
To be completely frank, I simply had not considered the idea that we could add more filtering on the API end! My mind was stuck in a modality of "the copy to the API needs to be API-results-ready", and I think having that data sit in the API DB and just filter it out in responses works great! I was quite hesitant about this section of the proposal because I don't like the idea of throwing data out either. This also makes me feel a lot better about how to handle the Clarifai data too.

Reply:
This is an interesting philosophical question about the catalog :) I'm inclined to agree with the characterization of the catalog as raw data that can be _filtered_ in the API, as long as we make a distinction between filtering and transforming data.

Totally separate to the cleanup steps, we already do this in the data refresh when we copy data into the API tables -- we [exclude rows](https://github.com/WordPress/openverse/blob/2da7be13b1c5dbca3d551b992fbbb90ec9be4cb9/ingestion_server/ingestion_server/queries.py#L178) that have an associated `DeletedMedia` record.